### PR TITLE
User story 14

### LINKED ISSUE #14: User Story 14
Image

As a merchant,
When I visit my merchant's invoices index (/merchants/:merchant_id/invoices)
Then I see all of the invoices that include at least one of my merchant's items
And for each invoice I see its id
And each id links to the merchant invoice show page

### PR DESCRIPTION
**Pull request by MWoodshop**
# Describe the changes below:

- Adds view file for merchant invoices.
- Moves invoices_controller out of Merchants directory.
- Adds index method to invoices_controller.
- Adds invoices method to merchant model.
- Adds merchant_invoices_index_spec.rb file in features directory.
- Adds two tests for User Story 17 to merchant_invoices_index_spec test file in features.

# Relevant user story:
As a merchant,
When I visit my merchant's invoices index (/merchants/:merchant_id/invoices)
Then I see all of the invoices that include at least one of my merchant's items
And for each invoice I see its id
And each id links to the merchant invoice show page

# Specific feedback requests? Include file name and line:
None specific, I believe Dani also moved her controller out of the subdirectory so there shouldn't be any issues.

# Before submitting, check the following:
- [X] Entire test suite is passing
- [X] SimpleCov test percentage

# Note any co-authors / anything else you'd like to add:

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+      <nav class="main-menu">
+        <%= link_to "Dashboard", root_path %>
+        <%= link_to "My Items", root_path %>
+        <%= link_to "My Invoices", merchant_invoices_path(@merchant) %>
+      </nav>
+    </div>
+    <div class="seperator"></div>
+    <div class="container">
+      <%= yield %>
+    </div>
```
Comment (westonio) [inline app/views/layouts/application.html.erb]: This looks good, but will cause everything to appear on all pages across the app and these nav links are not relevant for the Admin pages.

Can you move this over into a partial or your view's direct file instead?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
-    <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_importmap_tags %>
   </head>
-
   <body>
-    <%= yield %>
+    <h1 class="main_title">Welcome to Little Esty Shop</h1>
+    <div class="merchant_title">
+      <h2>Ray's Handmade Jewelry</h2>
```
Comment (westonio) [inline app/views/layouts/application.html.erb]: Also should the name of the merchant be dynamic?

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -13,3 +13,55 @@
...
+    text-align: center;
+    padding: 10px 0;
+    margin: 0;
+    font-size: 1.5rem;
+    font-weight: bold;
+}
+
+ul {
+  list-style-type: none;
+}
```
Comment (westonio) [inline app/assets/stylesheets/application.css:67]: This is all so beautiful 🤩

File: app/views/invoices/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="page-title">My Invoices</div>
+<ul>
+  <% @invoices.each do |invoice| %>
+    <li id="invoice-<%= invoice.id %>">
```
Comment (westonio) [inline app/views/invoices/index.html.erb:16]: Love the dynamic css ids here!

**Review by westonio — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (westonio) [reply app/views/layouts/application.html.erb]: Also, I believe Dani's User Stories include creating these nav links for the merchant pages. So you might want to double check with her on this aspect

Comment (MWoodshop) [reply app/views/layouts/application.html.erb]: @westonio I made the requested changes regarding the navbar. The CSS remains to be implemented sitewide later.

Comment (westonio) [reply app/views/layouts/application.html.erb]: Thank you kindly @MWoodshop ! looks perfect

**Review by westonio — APPROVED** [review]

File: spec/features/merchant_invoices_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  # And each id links to the merchant invoice show page
+  it 'displays all of the invoices that include at least one of my merchants items' do
+    visit merchant_invoices_path(@merchant1)
+
+    expect(page).to have_content(@invoice1.id)
+  end
+
+  it 'links each invoice id to the merchant invoice show page' do
+    visit merchant_invoices_path(@merchant1)
+    within "#invoice-#{@invoice1.id}" do
```
Comment (awiley33) [inline spec/features/merchant_invoices_index_spec.rb:29]: great use of within blocks here, it makes the test really robust

**Review by awiley33 — APPROVED** [review]
Nice work!

**Comment by MWoodshop** [comment]
@awiley33 looks like there are some merge conflicts here. Not sure if they send you an email about that but you have enough reviewers 👍 .

**Pull request merged**